src/bin/cql-stress-cassandra-stress/operation/mod.rs

```diff
@@ -16,6 +23,121 @@ use scylla::{
 };
 pub use write::WriteOperationFactory;
 
+use crate::settings::CassandraStressSettings;
+use crate::stats::ShardedStats;
+
+use self::row_generator::RowGenerator;
+
+/// A specific CassandraStress operation.
+#[trait_variant::make(CassandraStressOperation: Send)]
+pub trait LocalCassandraStressOperation: Sync + Send + Sized {
+    type Factory: CassandraStressOperationFactory<Operation = Self>;
+
+    const DEFAULT_TABLE_NAME: &'static str = "standard1";
+    const DEFAULT_COUNTER_TABLE_NAME: &'static str = "counter1";
+
+    fn get_stressed_table_name() -> &'static str;
+    async fn execute(&self, row: &[CqlValue]) -> Result<ControlFlow<()>>;
+    fn generate_row(&self, row_generator: &mut RowGenerator) -> Vec<CqlValue>;
+}
+
+pub trait CassandraStressOperationFactory: Sync + Send + Sized {
+    type Operation: CassandraStressOperation<Factory = Self>;
+
+    async fn new(settings: Arc<CassandraStressSettings>, session: Arc<Session>) -> Result<Self>;
```

Comment:
Do we actually need this to be a trait method? I think that `GenericCassandraStressOperationFactory` could just have a `new` method which takes something that implements `CassandraStressOperationFactory`. I don't think it will inflate code much, and I suppose that it will make it easier to get rid of `get_stressed_table_name` - for the read operation factory, you could have struct field that defines the table name.

Reply:
Yep, I see your point - makes sense to me.